txscript/standard.go

```diff
@@ -350,6 +350,21 @@ func PayToAddrScript(addr btcutil.Address) ([]byte, error) {
 	return nil, ErrUnsupportedAddress
 }
 
+// NullDataScript creates a provably prunable script
+// containing OP_RETURN followed by the passed data.
+func NullDataScript(data []byte) ([]byte, error) {
+	if len(data) > MaxDataCarrierSize {
+		return nil, ErrStackLongScript
+	}
+
+	script := make([]byte, 3+len(data))
```

Comment:
Another case would be:

`txscript.NullDataScript(bytes.Repeat([]byte{0x01, 24))`

Your code would produce `OP_RETURN OP_PUSHDATA1 0x18 0x01010101...` instead of `OP_RETURN OP_DATA_24 0x01010101...`.